### PR TITLE
ds-array in RandomForest

### PR DESCRIPTION
**Pull request by salvisolamartinell**
# Description
Added ds-array to RandomForest.

### CODE REVIEW

File: dislib/classification/rf/forest.py [hunk 1]
```diff
@@ -77,28 +77,31 @@ def __init__(self,
         self.hard_vote = hard_vote
         self.random_state = check_random_state(random_state)
 
-    def fit(self, dataset):
+    def fit(self, x, y, rf_dataset=False):
```
Comment (javicid) [inline dislib/classification/rf/forest.py]: Is rf_dataset necessary?

Comment (salvisolamartinell) [reply dislib/classification/rf/forest.py]: removed

File: dislib/classification/rf/forest.py [hunk 2]
```diff
@@ -122,77 +125,102 @@ def fit(self, dataset):
...
-        for subset in dataset:
+        probabilities = []
+        for x_row in x._iterator(axis=0):
             tree_predictions = []
             for tree in self.trees:
-                tree_predictions.append(tree.predict_proba(subset))
-            _join_predictions(subset, *tree_predictions)
+                tree_predictions.append(tree.predict_proba(x_row))
+            probabilities.append(_join_predictions(*tree_predictions))
+        return probabilities
```
[the remark this answers is not in the document]
Comment (salvisolamartinell) [reply dislib/classification/rf/forest.py]: Added synchronization and returning ds-array

File: dislib/classification/rf/forest.py [hunk 3]
```diff
@@ -122,77 +125,102 @@ def fit(self, dataset):
         for tree in self.trees:
             tree.fit(dataset)
 
-    def fit_predict(self, dataset):
-        """Fits the forest and predicts the classes for the same dataset.
+    def fit_predict(self, x, y):
```
[the remark this answers is not in the document]
Comment (salvisolamartinell) [reply dislib/classification/rf/forest.py]: Removed. Same can be done using fit().predict().

**Pull request merged**